### PR TITLE
[Docs] Updating contributing guide and codespaces configuration

### PR DESCRIPTION
**Pull request by antonymilne**
## Description

I'd like reviews here from @stichbury and @maxschulz-COL please 🙏  And then when it's published let's make sure everyone on the team reads the new guide.

In preparation for GHC, but also long overdue and something I've wanted to do for ages... I think we may have a ticket about this somewhere that we can hopefully now close @stichbury?

* update our codespaces configuration. It now runs much faster. This was a small change but took a lot of fiddling round to get the hatch install script etc. working correctly. @lingyielia and @maxschulz-COL should make sure this works and they can use it well enough to help other people use it at GHC!
* complete rewrite of the contributing guide. Don't try to look at the diff here, look at the built docs page instead: https://vizro--741.org.readthedocs.build/en/741/pages/explanation/contributing/
* update authors list
* move some questions from out contributing guide to the FAQ
* revert changes to schema generation in #738 since it was actually worse than the previous system

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/explanation/contributing.md [hunk 1]
```diff
@@ -1,192 +1,159 @@
...
+1. Use [Vizro's GitHub issues](https://github.com/mckinsey/vizro/issues) to discuss your contribution.
+2. [Create a codespace for our repository](https://codespaces.new/mckinsey/vizro). Leave the settings on their defaults and click "Create codespace" to start your codespace. It should take 1-2 minutes to fully launch and automatically start an example dashboard on port 8050.
+3. Make changes to Vizro code in your codespace. See the [GitHub Codespaces documentation on developing in a codespace](https://docs.github.com/en/codespaces/developing-in-a-codespace/developing-in-a-codespace) for more information.
+4. Add your name to the [list of contributors](authors.md) (source file `vizro-core/docs/pages/explanation/authors.md`).
+5. [Create a pull request (PR)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
+6. Before it can be merged, your PR must meet all the following requirements:
+    * Two approving reviews (including a code owner).
+    * Continuous Integration (CI) checks pass.
+    * Up-to-date with `main`.
```
Comment (stichbury) [inline vizro-core/docs/pages/explanation/contributing.md:31]: I'm tempted to take this out and put it up top so it applies to both methods of contributing, to avoid repetition.

Also, should you add something about the changelog requirement here too, and link through to the hatch command to add it?

File: vizro-core/docs/pages/explanation/contributing.md [hunk 2]
```diff
@@ -1,192 +1,159 @@
...
-If you are already active on the Plotly Community Forum, you can also ask your question over there in the [Dash Python category](https://community.plotly.com/c/python/25).
+!!! note
 
-## How to interact with the repository
+    There is no need to manually create a fork of the Vizro code if you use GitHub Codespaces, because fork is [automatically created for you](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-source-control-in-your-codespace#about-automatic-forking).
+    
+To develop on [GitHub Codespaces](https://docs.github.com/en/codespaces), follow the below steps:
 
-The easiest way to get up and running is to [open the repository in GitHub Codespaces](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=626855142). This creates a temporary development environment with all the necessary configurations, making it especially convenient for tasks like reviewing pull requests.
+1. Use [Vizro's GitHub issues](https://github.com/mckinsey/vizro/issues) to discuss your contribution.
```
Comment (stichbury) [inline vizro-core/docs/pages/explanation/contributing.md:23]: Wondering if this should be pulled out to the top to explain clearly how to find a ticket to work on and what to do before you get started if the work is more than a small tweak?

File: vizro-core/docs/pages/explanation/contributing.md [hunk 3]
```diff
@@ -1,192 +1,159 @@
...
+Run `hatch run changelog:add` to create a changelog fragment and then uncomment the relevant section(s). If you are uncertain about what to add or whether to add anything at all, refer to [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). The rule of thumb is that if Vizro users would be affected in any way then the changes should be described in the changelog.
 
-When executing the secret scan, there are two modes: `protect` can discover secrets in staged files, `detect` does so in the commit history.
+!!! note
 
-## Changelog
+    Changes that only affect documentation do not need a changelog fragment. This facilitates simple modifications to documentation [made directly on GitHub](https://docs.github.com/en/repositories/working-with-files/managing-files/editing-files) or with the [github.dev](https://docs.github.com/en/codespaces/the-githubdev-web-based-editor), where no terminal is available to run `hatch changelog:add`. Any changes outside documentation require a changelog fragment to be generated. If your changes do not require a changelog entry then you still need to generate the fragment but can leave it all commented out.
```
Comment (stichbury) [inline vizro-core/docs/pages/explanation/contributing.md:115]: I have never found it to be the case that I can get away with no changelog, for any docs change. Is this a recent improvement?

File: vizro-core/docs/pages/explanation/contributing.md (hunk 2 above)
Comment (stichbury) [reply vizro-core/docs/pages/explanation/contributing.md:23]: Maybe it's as simple as putting this big back that you removed (with some edit to add a bit about the requirements of a PR, which I've commented just below here)?

> Contributions of all experience levels are welcome! There are many ways to contribute, and we appreciate any help. Use our [issues page](https://github.com/mckinsey/vizro/issues) to discuss any contributions. Before opening a pull request, ensure you've first opened an issue to discuss the contribution.

File: vizro-core/docs/pages/explanation/contributing.md [hunk 4]
```diff
@@ -1,192 +1,159 @@
...
-## Got a Vizro question?
 
-We are happy to receive general questions around Vizro. Take a look at our [issues page](https://github.com/mckinsey/vizro/issues) and raise a ticket in the category `general question`. We would be grateful if you could check for any similar descriptions in the existing issues before opening a new ticket.
+## Develop on GitHub Codespaces
 
-If you are already active on the Plotly Community Forum, you can also ask your question over there in the [Dash Python category](https://community.plotly.com/c/python/25).
+!!! note
 
-## How to interact with the repository
+    There is no need to manually create a fork of the Vizro code if you use GitHub Codespaces, because fork is [automatically created for you](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-source-control-in-your-codespace#about-automatic-forking).
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/explanation/contributing.md:19]:
```suggestion
    There is no need to manually create a fork of the Vizro code if you use GitHub Codespaces, because a fork is [automatically created for you](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-source-control-in-your-codespace#about-automatic-forking).
```

As I said in catchup - this actually was not 100% without friction. It did work, but there was an error message.

File: vizro-core/docs/pages/explanation/contributing.md (hunk 2 above)
Comment (stichbury) [reply vizro-core/docs/pages/explanation/contributing.md:23]: Also, I think it would help to edit slightly, here's my suggestion for the first sentence:

```suggestion
Contributions of all experience levels are welcome! There are many ways to contribute, and we appreciate any help: it doesn't have to be a pull request on our code. You can also raise an issue to suggest an improvement to Vizro or our documentation, request a feature, or ask and answer community questions. See the FAQ for guidance on how to do these things.

<Then continue about the issues page blah blah>
```

File: vizro-core/docs/pages/explanation/contributing.md (hunk 1 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md:31]: Should we also mention that some CI needs to be activated?

**Review by stichbury — APPROVED** [review]
Few minor improvement suggestions. Plus a made a number of very small commits (individual sections so you can roll back any you disagree with) for minor tweaks.

File: vizro-core/docs/pages/explanation/contributing.md (hunk 2 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md:23]: @stichbury Do we use punctuation at the end of itemized lists?

File: vizro-core/docs/pages/explanation/contributing.md [hunk 5]
```diff
@@ -1,192 +1,159 @@
...
+    * Your fork is up-to-date with `main`.
 
-    If you haven't used Hatch before, it's well worth skimming through [their documentation](https://hatch.pypa.io/), in particular the page on [environments](https://hatch.pypa.io/latest/environment/). Run `hatch env show` to show all Hatch's environments and available scripts, and take a look at [`hatch.toml`](https://github.com/mckinsey/vizro/tree/main/vizro-core/hatch.toml) to see our Hatch configuration. [Hatch's tab completion](https://hatch.pypa.io/latest/cli/about/#tab-completion) is useful to explore the Hatch CLI.
+## How to use Hatch
 
----
+Regardless of whether you are developing locally or in a codespace, everything you need to develop on Vizro is provided by Hatch through the [`hatch run`](https://hatch.pypa.io/latest/cli/reference/#hatch-run) command. The first time you use this command it will install all the required dependencies, including Python, which may take some time.
 
-## Contribute to documentation
+The Hatch commands you need most commonly are as follows. These must be executed with `vizro-core` as your current working directory:
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/explanation/contributing.md:51]:
```suggestion
The Hatch commands you need most commonly are as follows. These must be executed with `vizro-core` as your current working directory (or `vizro-ai` if your are working on our that package):
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 6]
```diff
@@ -1,192 +1,159 @@
...
-
-## Code of conduct
-
-The Vizro team pledges to foster and maintain a friendly community. We enforce a [Code of Conduct](https://github.com/mckinsey/vizro/tree/main/CODE_OF_CONDUCT.md) to ensure every Vizro contributor is welcomed and treated with respect.
+```console
+hatch run all.py3.10:test-unit-coverage
+```
 
-## Frequently asked contribution questions
+In addition to unit tests with code coverage, CI also performs the following checks:
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/explanation/contributing.md:133]:
```suggestion
In addition to run unit tests with code coverage, CI also performs the following checks:
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 7]
```diff
@@ -1,192 +1,159 @@
...
 
-<!-- vale off -->
-### How do I add a dependency?
-<!-- vale on -->
+* `hatch run test-integration` runs integration tests that includes checking that the example apps in `vizro-core/examples` run.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/explanation/contributing.md:135]:
```suggestion
* `hatch run test-integration` runs integration tests that include checking that the example apps in `vizro-core/examples` run.
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 8]
```diff
@@ -1,192 +1,159 @@
...
 
-- `hatch new` template
-- [`copier-pylib`](https://github.com/astrojuanlu/copier-pylib)
-- [`cookiecutter-hypermodern-python`](https://github.com/cjolowicz/cookiecutter-hypermodern-python) and associated [user guide](https://cookiecutter-hypermodern-python.readthedocs.io/)
-- [`scikit-hep/cookie`](https://github.com/scikit-hep/cookie) and associated [developer guidelines](https://scikit-hep.org/developer)
+If for some reason you do need to use `pip` then the correct way to do so is through `hatch run pip`. For example, you could run `hatch run pip show plotly`. This will use the version of uv that Hatch itself uses under the hood. If you already have uv installed globally then `uv pip show plotly` would also work.
 
-Further useful articles:
+You _should not_ try to interact with Vizro dependencies using a global `pip`. For example, running `pip show plotly` without the `hatch run` prefix will not work correctly.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/explanation/contributing.md:155]:
```suggestion
!!! warning
    You _should not_ try to interact with Vizro dependencies using a global `pip`. For example, running `pip show 
    plotly` without the `hatch run` prefix will not work correctly.
```

File: vizro-core/mkdocs.yml [hunk 9]
```diff
@@ -98,6 +98,7 @@ markdown_extensions:
   - pymdownx.arithmatex:
       generic: true
   - footnotes
+  - pymdownx.keys
```
Comment (maxschulz-COL) [inline vizro-core/mkdocs.yml:101]: What is this for?

File: vizro-core/schemas/generate.py [hunk 10]
```diff
@@ -17,7 +17,10 @@
 
 if args.check:
     if json.loads(schema_path.read_text()) != json.loads(schema_json):
-        sys.exit("JSON schema is out of date. Run `hatch run schema` to update it.")
+        # Ideally just doing hatch run:schema would be fine, but the schema slightly depends
+        # on Python version (Python 3.8 vs. Python 3.11 give different results), even
+        # for the same pydantic version.
+        sys.exit("JSON schema is out of date. Run `hatch run all.py3.11:schema` to update it.")
```
Comment (maxschulz-COL) [inline vizro-core/schemas/generate.py:23]: Can we not do a double aliasing so that `hatch run schema` would actually run `hatch run all.py3.11:schema`. That seems easier no?

**Review by maxschulz-COL — COMMENTED** [review]
😍  Wow, really cool and streamlined. I think this sets up perfectly for upcoming contributions. It is now very fun to read!

Will approve once we get to the bottom of the strange codespace failure that happens sometimes.

What shall we do if we cannot ressolve this before GHC?

File: vizro-core/docs/pages/explanation/contributing.md (hunk 2 above)
Comment (stichbury) [reply vizro-core/docs/pages/explanation/contributing.md:23]: It depends. Typically we do if the bullets comprise one or more full sentence, but not if they're not because they're just statements. So it's like this pair of examples...

Some of the features include:
* feature 1
* feature 2
* feature 3

You need to consider the following:
* Do you really want to hurt me?
* Do you really want to make me cry?
* Do you know any other song lyrics that are questions?

(but it doesn't need to be a question, could just be a period as long as the text is a full and complete sentence).


HOWEVER, I don't always adhere to this and it can be difficult because sometimes bullets are not always consistent in the list itself (which could make it look weird) and sometimes there's another bulleted list nearby on the page that should be punctuated so it looks odd if there's one that is and one that isn't. 

TL;DR: I'd typically say, go with punctuation more times than not, and if in doubt, add it.

File: vizro-core/docs/pages/explanation/contributing.md (hunk 4 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:19]: Indeed... I would be tempted to leave these instructions as they are here (assuming the automatic fork works) and then you show the other flow where you do a manual fork in your GHC. wdyt?

File: vizro-core/docs/pages/explanation/contributing.md [hunk 11]
```diff
@@ -9,45 +9,46 @@ There are two ways to develop on Vizro:
...
-4. Make changes in your fork.
+3. [Fork the Vizro repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) and clone it to your local machine.
+4. Make changes to Vizro code in your fork.
 5. Add your name to the [list of contributors](authors.md) (source file `vizro-core/docs/pages/explanation/authors.md`).
 6. [Create a pull request (PR)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
 7. Before it can be merged, your PR must meet all the following requirements:
     * Two approving reviews (including a code owner).
     * Continuous Integration (CI) checks pass.
-    * Up-to-date with `main`.
+    * Your fork is up-to-date with `main`.
```
Comment (antonymilne) [inline vizro-core/docs/pages/explanation/contributing.md:45]:
```suggestion
    * Code is up-to-date with `main`.
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 12]
```diff
@@ -9,45 +9,46 @@ There are two ways to develop on Vizro:
...
+1. Use [Vizro's GitHub issues](https://github.com/mckinsey/vizro/issues) to discuss your contribution.
+2. [Create a codespace for our repository](https://codespaces.new/mckinsey/vizro). Leave the settings on their defaults and click "Create codespace" to start your codespace. It should take 1-2 minutes to fully launch and automatically start an example dashboard on port 8050.
+3. Make changes to Vizro code in your codespace. See the [GitHub Codespaces documentation on developing in a codespace](https://docs.github.com/en/codespaces/developing-in-a-codespace/developing-in-a-codespace) for more information.
 4. Add your name to the [list of contributors](authors.md) (source file `vizro-core/docs/pages/explanation/authors.md`).
 5. [Create a pull request (PR)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
 6. Before it can be merged, your PR must meet all the following requirements:
     * Two approving reviews (including a code owner).
     * Continuous Integration (CI) checks pass.
     * Up-to-date with `main`.
```
Comment (antonymilne) [inline vizro-core/docs/pages/explanation/contributing.md:31]:
```suggestion
    * Code is up-to-date with `main`.
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 13]
```diff
@@ -9,45 +9,46 @@ There are two ways to develop on Vizro:
...
 7. Before it can be merged, your PR must meet all the following requirements:
     * Two approving reviews (including a code owner).
     * Continuous Integration (CI) checks pass.
-    * Up-to-date with `main`.
+    * Your fork is up-to-date with `main`.
 
 ## How to use Hatch
 
-Regardless of whether you are developing locally or in a codespace, everything you need to develop on Vizro is provided by Hatch through the [`hatch run`](https://hatch.pypa.io/latest/cli/reference/#hatch-run) command. The first time you use this command it will install all the required dependencies, including Python.
+Regardless of whether you are developing locally or in a codespace, everything you need to develop on Vizro is provided by Hatch through the [`hatch run`](https://hatch.pypa.io/latest/cli/reference/#hatch-run) command. The first time you use this command it will install all the required dependencies, including Python, which may take some time.
```
Comment (antonymilne) [inline vizro-core/docs/pages/explanation/contributing.md:49]:
```suggestion
Regardless of whether you are developing locally or in a codespace, everything you need to develop on Vizro is provided by Hatch through the [`hatch run`](https://hatch.pypa.io/latest/cli/reference/#hatch-run) command. The first time you use this command it will install all the required dependencies, including Python.
```

Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:49]: FYI @stichbury thanks to uv this is actually now ridiculously fast (like <30s) 😀 The only slow part of our local development process now is initially running linting.

File: vizro-core/docs/pages/explanation/contributing.md [hunk 14]
```diff
@@ -135,13 +138,13 @@ In addition to unit tests with code coverage, CI also performs the following che
 
 ### `hatch run docs:serve`
 
-`hatch run docs:serve` shows hot-reloading documentation. Documentation is also built automatically in your PR and can be previewed on Read The Docs. To do this, scroll to the bottom of your PR where all the checks are listed and click the "Details" link next to the Read the Docs build.
+`hatch run docs:serve` builds and displays documentation that is hot-reloading for convenience while you edit it. Documentation is also built automatically in your PR and can be previewed on Read The Docs. To do this, scroll to the bottom of your PR where all the checks are listed and click the "Details" link next to the Read the Docs build.
```
Comment (antonymilne) [inline vizro-core/docs/pages/explanation/contributing.md:141]:
```suggestion
`hatch run docs:serve` builds and displays documentation that hot-reloads while you edit it. Documentation is also built automatically in your PR and can be previewed on Read The Docs. To do this, scroll to the bottom of your PR where all the checks are listed and click the "Details" link next to the Read the Docs build.
```

File: vizro-core/docs/pages/explanation/contributing.md [hunk 15]
```diff
@@ -135,13 +138,13 @@ In addition to unit tests with code coverage, CI also performs the following che
...
 
 For more information on our documentation style, refer to our [style guide](documentation-style-guide.md).
 
 ### `hatch run pip`
 
-`hatch run pip` provides a [pip-compatible interface using uv](https://docs.astral.sh/uv/pip/). You should not need to use this much.
+`hatch run pip` provides a [pip-compatible interface using uv](https://docs.astral.sh/uv/pip/). You should not need to use this.
```
Comment (antonymilne) [inline vizro-core/docs/pages/explanation/contributing.md:147]:
```suggestion
`hatch run pip` provides a [pip-compatible interface using uv](https://docs.astral.sh/uv/pip/). You should not need to use this often.
```

File: vizro-core/mkdocs.yml (hunk 9 above)
Comment (antonymilne) [reply vizro-core/mkdocs.yml:101]: It makes the `++ctrl+shift+p++` render as keyboard keys.

File: .devcontainer/install-hatch.sh [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+RUNNER_OS="Linux"
+RUNNER_ARCH="X64"
+
+set -euo pipefail
+IFS=$'\n\t'
+
+PURPLE="\033[1;35m"
+RESET="\033[0m"
+
+if [[ "${RUNNER_OS}" == "Windows" ]]; then
```
[the remark this answers is not in the document]
Comment (antonymilne) [reply .devcontainer/install-hatch.sh:16]: For us it wouldn't. This script is basically copied and pasted from https://github.com/pypa/hatch/tree/install and I didn't want to modify anything more than the minimum to get it working for us (so variables like `RUNNER_OS` are hardcoded at the top).

File: vizro-core/schemas/generate.py (hunk 10 above)
Comment (antonymilne) [reply vizro-core/schemas/generate.py:23]: We could indeed do this. It's kind of ugly because doing `hatch run schema` then actually creates both the `default` environment and then the `all.py3.11` inside that environment. So it's a pretty nasty hack but it does seem to work ok for `hatch run example`, so let me change it here - good idea!

Comment (antonymilne) [reply vizro-core/schemas/generate.py:23]: Ah, also the reason I didn't do this before was because `hatch prep-release` runs from `default` environment, and so it would have been much slower to run that because it would have needed to create the `all.py3.11` environment half way through that. But now that environment creation is so fast this is all good 🎉

File: vizro-core/docs/pages/explanation/contributing.md (hunk 3 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:115]: Not that recent: https://github.com/mckinsey/vizro/pull/480.

I did test it pretty thoroughly at the time and thought it seemed to work well, so if it's not working for you then please do point me towards a PR when you next find that it's not working as intended!

File: vizro-core/docs/pages/explanation/contributing.md (hunk 1 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md]: @stichbury yeah, I wasn't sure about this. Given that a contributor would be using either the Codespace route or the local development route I thought it would be easier for someone to just follow all the steps in one of those sections without having to jump between a separate section that contains the parts that are common to both those processes and then back again. But it definitely makes sense to pull out point 1 on the list of both to the top of the page, so let me try and move this last point there as well and see if it reads ok.

I decided to not add anything about the changelog requirement here because:
* it's included in the "Continuous Integration (CI) checks pass" point
* when that check fails, the error message is very clear and easy to follow since it instructs the user to run `hatch run changelog:add`
* in theory changelogs aren't needed for every change anyway (purely docs changes shouldn't require it - if that's not the case then it's not working right)

@maxschulz-COL what do you mean by activated? Do the required jobs for external contributors not get triggered without some intervention?

File: vizro-core/docs/pages/explanation/contributing.md (hunk 2 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:23]: This is the sort of guide I was trying to follow so good to know, thanks @stichbury!

And good suggestions with these edits, I will definitely change as you suggest.

File: .devcontainer/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,8 @@
+FROM mcr.microsoft.com/devcontainers/base:debian
```
[the remark this answers is not in the document]
Comment (antonymilne) [reply .devcontainer/Dockerfile:1]: That's strange, I have no idea what's happening here. Did that happen when you started a devcontainer _from this branch_? Do you have the logs that go before that? 🤔 

FYI if the startup fails then it reverts to building the default docker image. So you can manually get to the "correct state" by manually running the code the Dockerfile should have done:
```
cd .devcontainer && chmod +x install-hatch.sh && ./install-hatch.sh && rm install-hatch.sh
echo "alias hr='hatch run'" >> ~/.bashrc
cd ../vizro-core && hr examples:example
```

Maybe I should add this to the contributor guide? Or is the failure rare enough that in such cases we should just tell people to start again? Does [rebuilding the container](https://docs.github.com/en/codespaces/developing-in-a-codespace/rebuilding-the-container-in-a-codespace#rebuilding-a-container) fix it? If so then that's probably the easiest fix.

File: vizro-core/docs/pages/explanation/contributing.md [hunk 18]
```diff
@@ -1,192 +1,156 @@
...
-4. Run `hatch run example` to [open an example Vizro dashboard](../examples/examples.md) with [Dash dev tools](https://dash.plotly.com/devtools) enabled.
-5. Edit the code! Thanks to Dash dev tools' hot reloading, any changes to the example app or `vizro` source code should automatically show in your dashboard without needing refresh or restart any process.
+To develop on [GitHub Codespaces](https://docs.github.com/en/codespaces), follow the below steps:
 
-!!!note
+1. [Create a codespace for our repository](https://codespaces.new/mckinsey/vizro). Leave the settings on their defaults and click "Create codespace" to start your codespace. It should take 1-2 minutes to fully launch and automatically start an example dashboard on port 8050.
+2. Make changes to Vizro code in your codespace. See the [GitHub Codespaces documentation on developing in a codespace](https://docs.github.com/en/codespaces/developing-in-a-codespace/developing-in-a-codespace) for more information.
```
Comment (lingyielia) [inline vizro-core/docs/pages/explanation/contributing.md:30]: Do we want to suggest contributors create a feature branch in codespace, and changes go to the feature branch of their fork?

File: vizro-core/docs/pages/explanation/contributing.md [hunk 19]
```diff
@@ -1,192 +1,156 @@
...
+3. Add your name to the [list of contributors](authors.md) (source file `vizro-core/docs/pages/explanation/authors.md`).
+4. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
```
Comment (lingyielia) [inline vizro-core/docs/pages/explanation/contributing.md:32]: Tested with another account, the process is straightforward 👍 

When raising the PR, one check is failing: `Labeler / label (pull_request) `

File: vizro-core/docs/pages/explanation/contributing.md (hunk 1 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md]: I was under the impression that yes. When I did my testing I wasn't focussing on that, but it looked to me like it. @l0uden do you know if there is any activation needed when raising a PR from a fork?

I can test later again though just to be sure

File: .devcontainer/install-hatch.sh (hunk 16 above)
[the remark this answers is not in the document]
Comment (maxschulz-COL) [reply .devcontainer/install-hatch.sh:16]: Hm ok - let's leave it then although it looked a bit confusing to me. Maybe add a comment?

File: vizro-core/docs/pages/explanation/contributing.md (hunk 4 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md]: Hm ok - let's do that. Do you then think it is an issue that I even show something different at GHC - I just don't want people to even encounter any error messages

File: vizro-core/docs/pages/explanation/contributing.md (hunk 18 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md:30]: Good point, I also explain that in GHC as well. But on second thought this is probably mostly good practice for code hygiene - the PR will anyway come from another branch, so it is not necessary.

I am inclined to not mention it here because it keeps things simpler? Not entirely sure though

File: .devcontainer/Dockerfile (hunk 17 above)
[the remark this answers is not in the document]
Comment (maxschulz-COL) [reply .devcontainer/Dockerfile:1]: Today it created it without issue... Given my above findings (for the overall contribution process) I am inclined to leave as is, as there seem to be always bugs sometimes. Maybe add a comment saying: "If something goes wrong, try re-creating the codespace, or building codespace from a fork created manually". But I would not post the above commands - that would be overkill. Wdyt?

File: vizro-core/docs/pages/explanation/contributing.md (hunk 4 above)
Comment (maxschulz-COL) [reply vizro-core/docs/pages/explanation/contributing.md]: As mentioned above, I am now not creating a fork first for GHC, but let's be aware that there sometimes can be issues..

**Review by maxschulz-COL — APPROVED** [review]
I will approve now pending the open comments 💪

File: vizro-core/docs/pages/explanation/contributing.md (hunk 19 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:32]: Good spot! I've just pushed something that will hopefully fix this but would be great to check again.

File: vizro-core/docs/pages/explanation/contributing.md (hunk 1 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md]: 1. Indeed seems like the automatic forking is not 100% reliable... but if it's working for you now then agreed with your proposal - let's leave the contribution guide as is then 👍 
2. Ah yes, the reason for this is this setting (which we currently have to be as permissive as possible). I will add a mention to this to the guide, good spot!
image

Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md]: @maxschulz-COL this is well worth knowing for GHC actually, given that many of them might have new github accounts so fit into this category.

File: vizro-core/docs/pages/explanation/contributing.md (hunk 18 above)
Comment (antonymilne) [reply vizro-core/docs/pages/explanation/contributing.md:30]: I think it's fine for a PR from a fork to come from whatever branch they like. I'd only do a branch on a fork if I was working on multiple things in parallel. So I think fine to not mention here and just keep things simple.

**Review by lingyielia — APPROVED** [review]
